shared/actions/chat/index.js

```diff
@@ -1234,12 +1238,12 @@ function* _updateTempSearchConversation(
   yield all(actionsToPut)
 }
 
-function* _exitSearch() {
-  const inboxSearch = yield select(inboxSearchSelector)
+function* _exitSearch({payload: {forSelectConversation}}: Constants.ExitSearch) {
```

Comment:
I would change this to be `skipSelectPreviousConversation: boolean` because that's what it really does. Then I would just check skipSelectPreviousConversation in L1246

Reply:
Changed the name, but unfortunately flow isn't smart enough to deduce that `!skipSelectPreviousConversation` implies `inboxSearch` is non-null